### ISSUE TITLE
[Shell] Unable to focus search handler

### ISSUE THREAD

**Issue by davidortinau** [xamarin/Xamarin.Forms#5768]
### Description

https://www.screencast.com/t/kmwfyFkTdS

It's possible to show and hide the search box, however you cannot set focus. I don't see any method for focus or to get the Entry field to set focus myself.

I suspect this will be an accessibility problem.

### Steps to Reproduce

1. Start with search hidden
2. Tap a search icon (toolbaritem or whatever) to open the search (expanded)
3. Observe there is not way to set focus programmatically. 

### Expected Behavior

Can set focus automatically OR explicitly.

### Actual Behavior

Cannot. 

### Basic Information

- Version with issue: 4.0-pre8

### Reproduction Link

You can use this if the gallery doesn't already have tests for it. https://github.com/davidortinau/Gastropods
